lex/lexer.go

```diff
@@ -644,8 +649,7 @@ func (l *Lexer) isExpr() bool {
 
 // non-consuming check to see if we are about to find next keyword
 func (l *Lexer) isNextKeyword(peekWord string) bool {
-
-	if len(peekWord) == 0 {
+	if len(peekWord) == 0 || l.curClause == nil {
```

Comment:
`curClause` appears to be the only state variable of `Lexer` that was not safe to use with its default value.